services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/NotificationManagerImpl.java

```diff
@@ -0,0 +1,58 @@
+package org.sagebionetworks.repo.manager;
+
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.util.Set;
+
+import org.apache.commons.fileupload.FileItemStream;
+import org.apache.http.entity.ContentType;
+import org.sagebionetworks.repo.manager.file.FileHandleManager;
+import org.sagebionetworks.repo.model.UserInfo;
+import org.sagebionetworks.repo.model.file.FileHandle;
+import org.sagebionetworks.repo.model.message.MessageToUser;
+import org.sagebionetworks.repo.web.NotFoundException;
+import org.sagebionetworks.repo.web.ServiceUnavailableException;
+import org.springframework.beans.factory.annotation.Autowired;
+
+public class NotificationManagerImpl implements NotificationManager {
+	
+	@Autowired
+	private FileHandleManager fileHandleManager;
+	
+	@Autowired
+	private MessageManager messageManager;
+
+	private static final Charset DEFAULT_CHARSET = Charset.defaultCharset();
+	
+	public NotificationManagerImpl() {}
+	
+	public NotificationManagerImpl(
+			FileHandleManager fileHandleManager, 
+			MessageManager messageManager) {
+		this.fileHandleManager = fileHandleManager;
+		this.messageManager = messageManager;
+	}
+	
+	@Override
+	public void sendNotification(UserInfo userInfo, MessageToUser mtu,
+			String message) throws NotFoundException {
+		
+		if (mtu.getRecipients().isEmpty()) return;
+
+		ContentType contentType = ContentType.create(NotificationManager.TEXT_PLAIN_MIME_TYPE, DEFAULT_CHARSET);
+		FileItemStream fileItemStream = new ByteArrayFileItemStream(
+					message.getBytes(DEFAULT_CHARSET), contentType.toString(), "");
+		try {
+			FileHandle fileHandle = fileHandleManager.uploadFile(userInfo.getId().toString(), fileItemStream);
```

Comment:
filehandlemanager.uploadFile() should be deprecated and removed.  An alternative that also compresses is: createCompressedFileFromString()  also need a readCompressedStringFromFileHandle();  Also remove the write transaction from the createCompressedFileFromString()